cmd/handler-utils.go

```diff
@@ -174,6 +177,45 @@ func extractMetadataFromMap(ctx context.Context, v map[string][]string, m map[st
 	return nil
 }
 
+// extractTags extracts tag key and value from given http header. It then
+// - Parses the input format X-Amz-Tagging:"Key1=Value1&Key2=Value2" into a map[string]string
+// with entries in the format X-Amg-Tag-Key1:Value1, X-Amz-Tag-Key2:Value2
+// - Validates the tags
+// - Returns the Tag in original string format "Key1=Value1&Key2=Value2"
+func extractTags(ctx context.Context, tags string) (string, error) {
+	// Check if the metadata has tagging related header
+	if tags != "" {
+		tagging, err := extractTagsFromString(tags)
+		if err != nil {
+			return "", err
+		}
+		if err := tagging.Validate(); err != nil {
+			return "", err
+		}
+		return tagging.String(), nil
+	}
+	return "", nil
+}
+
+func extractTagsFromString(tagStr string) (tagging.Tagging, error) {
+	tags, err := url.ParseQuery(tagStr)
+	if err != nil {
+		return tagging.Tagging{}, err
+	}
+	var idx = 0
+	parsedTags := make([]tagging.Tag, len(tags))
+	for k, v := range tags {
+		parsedTags[idx].Key = k
+		parsedTags[idx].Value = strings.Join(v, ",")
```

Comment:
Each tag has only one value so we don't need to use strings.Join instead just tags.Get(k) would work @nitisht